src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift

```diff
@@ -0,0 +1,189 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import OpenCombine
+import HealthCertificateToolkit
+
+class HealthCertificateService: HealthCertificateServiceProviding {
+
+	// MARK: - Init
+
+	init(
+		store: HealthCertificateStoring
+	) {
+		self.store = store
+
+		updatePublishersFromStore()
+		updateHealthCertifiedPersonSubscriptions()
+		updateProofCertificateOnDidBecomeActive()
+	}
+
+	// MARK: - Internal
+
+	private(set) var healthCertifiedPersons = CurrentValueSubject<[HealthCertifiedPerson], Never>([]) {
+		didSet {
+			store.healthCertifiedPersons = healthCertifiedPersons.value
+
+			updateHealthCertifiedPersonSubscriptions()
+		}
+	}
+
+	func registerHealthCertificate(
+		base45: Base45
+	) -> Result<HealthCertifiedPerson, HealthCertificateServiceError.RegistrationError> {
+		Log.info("[HealthCertificateService] Registering health certificate from payload: \(private: base45)", log: .api)
+
+		do {
+			let healthCertificate = try HealthCertificate(base45: base45)
+
+			guard let vaccinationCertificate = healthCertificate.vaccinationCertificates.first else {
+				return .failure(.noVaccinationEntry)
+			}
+
+			let healthCertifiedPerson = healthCertifiedPersons.value.first ?? HealthCertifiedPerson(healthCertificates: [], proofCertificate: nil)
```

Comment:
Instead of picking the first one. Could we filter for one with the right name and dateOfBirth?

Reply:
Ok, just wondering how it will work if we, maybe in future, have more than one certified person.
Then we somehow need to identify those persons. And if the identification is over name and dateOfBirth, I wonder how to have errors for wrong name or dateOfBirth.

Reply:
I guess if we support that in the future we will throw these errors out then and just register a new health certified person if none with matching name and date of birth exists.